staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4105,23 +4128,13 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinityConfig *SessionAffinityConfig `json:"sessionAffinityConfig,omitempty" protobuf:"bytes,14,opt,name=sessionAffinityConfig"`
 
-	// ipFamily specifies whether this Service has a preference for a particular IP family (e.g.
-	// IPv4 vs. IPv6) when the IPv6DualStack feature gate is enabled. In a dual-stack cluster,
-	// you can specify ipFamily when creating a ClusterIP Service to determine whether the
-	// controller will allocate an IPv4 or IPv6 IP for it, and you can specify ipFamily when
-	// creating a headless Service to determine whether it will have IPv4 or IPv6 Endpoints. In
-	// either case, if you do not specify an ipFamily explicitly, it will default to the
-	// cluster's primary IP family.
-	// This field is part of an alpha feature, and you should not make any assumptions about its
-	// semantics other than those described above. In particular, you should not assume that it
-	// can (or cannot) be changed after creation time; that it can only have the values "IPv4"
-	// and "IPv6"; or that its current value on a given Service correctly reflects the current
-	// state of that Service. (For ClusterIP Services, look at clusterIP to see if the Service
-	// is IPv4 or IPv6. For headless Services, look at the endpoints, which may be dual-stack in
-	// the future. For ExternalName Services, ipFamily has no meaning, but it may be set to an
-	// irrelevant value anyway.)
-	// +optional
-	IPFamily *IPFamily `json:"ipFamily,omitempty" protobuf:"bytes,15,opt,name=ipFamily,Configcasttype=IPFamily"`
+	// IPFamilies identifies all the IPFamilies assigned to or requested for
+	// this Service. If value was not provided for IPFamilies it will be defaulted
+	// based on the cluster configuration and the value of Service.Spec.IPFamilyPolicy.
+	// Maximum of two values (dualstack IPFamilies) are allowed in IPFamilies.
+	// +listType=atomic
+	// +optional
+	IPFamilies []IPFamily `json:"ipFamilies,omitempty" protobuf:"bytes,15,opt,name=ipFamilies,casttype=IPFamily"`
```

Comment:
I think you are right. we need somebody with deeper experience in proto. We might get away with this if scalar looks like array[1] on the wire. I will double check.

Reply:
I think scalar and array[1] are wire compatible, but I still wouldn't do it.

Reply:
Both keeping as is, and changing it will work (assuming scalar => array[1]). Changing it will break existing clients, though. Which should be fine since we are at alpha stage. 

Change?